jstest/testrunner/devices/serial_device.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2018-present Samsung Electronics Co., Ltd. and other contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from jstest.common import console
+from jstest.testrunner import utils as testrunner_utils
+from jstest.testrunner.devices.device_base import RemoteDevice
+from jstest.testrunner.devices.connections.serialcom import SerialConnection
+
+
+class SerialDevice(RemoteDevice):
+    def __init__(self, env):
+        RemoteDevice.__init__(self, env)
+
+        data = {
+            'dev-id': env['info']['device_id'],
+            'baud': env['info']['baud'],
+            'timeout': env['info']['timeout']
+        }
+
+        if self.device == 'artik053':
+            data['prompt'] = 'TASH>>'
+        else:
+            data['prompt'] = 'nsh> '
+
+        self.channel = SerialConnection(data)
+
+    def check_args(self):
+        '''
+        Check that all the arguments are established.
+        '''
+        if not self.env['info']['device_id']:
+            console.fail('Please use the --device-id to select the device.')
+
+    def reset(self):
+        '''
+        Dummy method.
+        '''
+        pass
+
+    def execute(self, testset, test):
```

Comment:
This method does execute nothing. Instead, this is just a command preparer method for the subclasses. Could you rename this method to `prepare_command` (or something else) and return with the prepared command?